services/wallet/activity/filter.sql

```diff
@@ -157,6 +157,43 @@ layer2_networks(network_id) AS (
 mint_methods(method_hash) AS (
 	%s
 )
+-- Select all token transfers
+, token_rows AS (
+    SELECT *
+    FROM transfers
+    WHERE type != 'eth'
+)
+, eth_nonzero_value_rows AS (
+    SELECT *
+    FROM transfers
+    WHERE type = 'eth' AND amount_padded128hex != '00000000000000000000000000000000'
+)
+, eth_zero_value_rows AS (
+    SELECT *
+    FROM transfers
+    WHERE type = 'eth' AND amount_padded128hex = '00000000000000000000000000000000'
+)
+-- Select non-gas-only ETH transfers
+, eth_real_zero_value_rows AS (
+    SELECT *
+    FROM eth_zero_value_rows
+    WHERE (tx_hash, address, network_id, account_nonce) NOT IN (
+        SELECT tx_hash, address, network_id, account_nonce
+        FROM token_rows
+    )
+)
+-- Merge all transfers
+, clean_transfers AS (
```

Comment:
What is the performance impact of adding this filter layer on top of the `transfers` table?

We already have a considerable performance penalty with the complex query, that I could not find a way to optimize further. 
The last performance analysis conclusion was that the way to go is to cache as many or all the criteria, dynamically aggregated here, into an activity table. Hence, I'm thinking that the way to do this without a high performance impact is by adding an indexed flag of gas transfer and filter that out. What do you think @IvanBelyakoff would that be possible?

Reply:
> Do we have any settled way to measure it consistently to know how each PR affects performance?

There are these benchmarks that I used in the performance improvements, I'm not sure they work still https://github.com/status-im/status-go/blob/develop/services/wallet/activity/benchmarks_test.go
I had on my list to run them in CI. However, because there is nothing yet there and have to build a foundation, that was never a priority.

You can find the last results in the last commit message: https://github.com/status-im/status-go/commit/579f7e4a523b2734a589bc452d4f391334627a0c

One way to run them is by `go test -v ./services/wallet/activity/... --benchtime=10x '--run=^$' --bench=BenchmarkGetActivityEntries --tags=gowaku_no_rln,gowaku_skip_migrations"` that I found in my command history
```

Reply:
>  ... but since we have performance issues, I would consider it now

The performance slowdown is visible on a heavy history like the one I used to do the hands-on benchmarks using the `0x1` address which has many thousands of activity entries. A regular user won't notice it so there is no urgent concern.

Reply:
We can go with the current change but track the effects over time. AFAIK we don't yet have defined some reference user profiles to generate some performance base line and check it from time to time or regularly, therefore all my understanding is based on a lot of assumptions.